src/neynar-api/v2/client.ts

```diff
@@ -927,6 +932,50 @@ export class NeynarV2APIClient {
     return response.data;
   }
 
+/**
+ * Retrieves detailed information about a cast conversation based on a specified hash or URL. Useful
+ * for fetching in-depth details of a single cast conversation, including replies up to a specified depth. The method
+ * allows for specifying the type of the provided identifier (e.g., a hash or URL) and how deep the reply chain should
+ * be fetched.
+ *
+ * @param {string} castHashOrUrl - The hash or URL of the cast conversation to be retrieved.
+ * @param {CastParamType} type - Specifies the type of the provided identifier, indicating whether it's a hash or URL.
+ * @param {number} [replyDepth] - Optional parameter to specify how deep the reply chain should be fetched.
+ * @param {boolean} [includeChronologicalParentCasts] - Optional parameter to include chronological parent casts in the response.
+ * @returns {Promise<Conversation>} A promise that resolves to a `Conversation` object,
+ *   containing detailed information about the requested cast conversation, including replies up to the specified depth.
+ *
+ * @example
+ * // Fetch detailed information about a cast conversation using a given hash or URL, with a reply depth of 2
+ * client.lookupCastConversation(
+ *   'https://warpcast.com/rish/0x9288c1',
+ *   CastParamType.Url,
+ *   2,
```

Comment:
@MrKevinOConnell Optional parameters are not passed correctly it should be `{ replyDepth: 2, includeChronologicalParentCasts: true}`

Though this example will not be displayed anywhere rn, maybe in the future if we decide to expose version-specific methods then this example will be incorrect.